app/src/components/events/Recommend.tsx

```diff
@@ -0,0 +1,87 @@
+import { QueryDocumentSnapshot, collection, doc, getDocs, query, setDoc, where } from "firebase/firestore";
+import { db } from "../../config/firebase-config";
+import { useEffect, useState } from "react";
+import data from "./events.json"
+
+export function Recommend() {
+    const [events, setEvents] = useState<QueryDocumentSnapshot[]>();
+    const [histRating, setHistRating] = useState<number | undefined>();
+    const [value, setValue] = useState<number | undefined>();
+
+    function averageRating(collection: QueryDocumentSnapshot[] | undefined) {
+        var sum = 0;
+        
+        if (collection) {
+            collection.forEach(document => {
+                sum += average(document.data().ratings);
+              });
+            
+            setHistRating(sum / collection.length);
+            return (sum / collection.length);
+        } else {
+            return 0;
+        }
+      }
+    
+    const average = (array: number[]) => {
+        var sum = 0;
+    
+        array.forEach(element => {
+          sum += element;
+        });
+    
+        return (sum / array.length);
+      }
+    
+    const getAverageRating = async (field: string, option: string) => {
+        // Get the historical (filtered) event data
+        const eventsRef = collection(db, "Events");
+
+        const querySnap = await getDocs(query(eventsRef, where(field, "==", option)));
+    
+        setEvents(querySnap.docs)
```

Comment:
does this need to be a global const, or can you just pass in the `querySnap.docs` to `averageRating` on line 44 below?

Reply:
I agree, its different in most recent version